ORE_EEG_2.py

```diff
@@ -5,24 +5,180 @@
 
-def setup_eeg():
+async def setup_eeg():
+    
     # Initialize EEG, e.g., with Emotiv SDK
     # This function needs to be implemented based on your EEG SDK's documentation
-    pass
-
-
-def record_trigger(trigger_number, debug_mode=True):
+    async with websockets.connect("wss://localhost:6868", ssl=ssl_context) as websocket:
+        # by default emotiv uses port 6868, uses wss protocol
+        # first need to request access
+        await send_message({
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "requestAccess",
+            "params": {
+                "clientId": os.environ.get('CLIENT_ID'),
+                "clientSecret": os.environ.get('CLIENT_SECRET'),
+            }
+        }, websocket)
+        # give it access through launcher
+        # refresh the device list
+        await send_message({
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "controlDevice",
+            "params": {
+                "command": "refresh"
+            }
+        }, websocket)
+        # query the headsets
+        response = await send_message({
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "queryHeadsets"
+        }, websocket)
+        if len(response["result"]) == 0:
+            print("No headsets found")
+            return
+        # connect to the headset
+        headset = response["result"][0]["id"] # assuming the first headset, otherwise can manually specifiy
+        await send_message({
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "controlDevice",
+            "params": {
+                "command": "connect",
+                "headset": headset,
+                "mappings": { # under the assumption that the headset is an EPOC Flex
+                    "CMS": "F3",
+                    "DRL": "F5",
+                    "LA": "AF3",
+                    "LB": "AF7",
+                    "RA": "P8"
+                }
+            }
+        }, websocket)
+        response = await send_message({ # authorize the connection
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "authorize",
+            "params": {
+                "clientId": os.environ.get('CLIENT_ID'),
+                "clientSecret": os.environ.get('CLIENT_SECRET'),
+                "debit": 1000
+            }
+        }, websocket)
+        if "error" in response:
+            print(response["error"]) # if it gets here, probably didn't set up env variables correctly
+            return
+        cortex_token = response["result"]["cortexToken"]
+        response = await send_message({
+            "id": 1,
+            "jsonrpc": "2.0",
+            "method": "createSession",
+            "params": {
+                "cortexToken": cortex_token,
+                "headset": headset,
+                "status": "open"
+            }
+        }, websocket)
+        session_id = response["result"]["id"]
```

Comment:
how does the value of session_id change? Will it be different every time?

Reply:
yeah, after the script stops running the session is closed so a new session id needs to be generated on the next run

> A session is linked to an application. All the sessions of an application are automatically closed when the application is disconnected from the Cortex service. A session is also closed if the headset is disconnected.
[from docs](https://emotiv.gitbook.io/cortex-api/session)